src/qibo/core/cgates.py

```diff
@@ -336,10 +328,10 @@ def __init__(self, q, theta, trainable=True):
         MatrixGate.__init__(self)
         gates.RY.__init__(self, q, theta, trainable)
 
-    def construct_unitary(self) -> np.ndarray:
+    def construct_unitary(self):
         theta = self.parameters
-        cos, sin = np.cos(theta / 2.0), np.sin(theta / 2.0)
-        return np.array([[cos, -sin], [sin, cos]], dtype=DTYPES.get('NPTYPECPX'))
+        cos, sin = qnp.cos(theta / 2.0), qnp.sin(theta / 2.0)
```

Comment:
Ok, that's a good point. So ideally the code is trying to do:
- use math for constants and expected python numerical types
- use numpy (`from qibo import numpy`) for objects which must be or are expected to be numpy objects
- use K in places where objects rely on the typo of the state.

The idea is good, however in places like here, it supposes the parameters will be as numpy arrays or compatible formats (tf which casts to numpy) or python lists. I am fine with these choices, but would be great to have some place where we summarize how to use `K`, `qibo.numpy` and `math`.